apis/v1alpha1/gateway_types.go

```diff
@@ -481,36 +479,44 @@ type RouteBindingSelector struct {
 	Resource string `json:"resource"`
 }
 
+// GatewaySelectType specifies where Routes should be selected by a Gateway.
+// +kubebuilder:validation:Enum=AllNamespaces;Selector;SameNamespace
+// +kubebuilder:default=SameNamespace
+type GatewaySelectType string
```

Comment:
nit: I was a little confused. Should this be RouteSelectType instead?

Reply:
Good call, should be fixed now.